java/ingest/ingest-runner/src/test/java/sleeper/ingest/testutils/IngestTestHelper.java

```diff
@@ -220,4 +179,84 @@ public IngestTestHelper<T> partitionFileWriterFactory(PartitionFileWriterFactory
+
+        Map<Integer, List<Record>> partitionNoToExpectedRecordsMap = expectedRecords.stream()
+                .collect(Collectors.groupingBy(
+                        record -> keyToPartitionNoMappingFn.apply(Key.create(record.getValues(schema.getRowKeyFieldNames())))));
+
+        Map<String, List<FileInfo>> partitionIdToFileInfosMap = stateStore.getActiveFiles().stream()
+                .collect(Collectors.groupingBy(FileInfo::getPartitionId));
+
+        Map<String, Integer> partitionIdToPartitionNoMap = partitionNoToExpectedRecordsMap.entrySet().stream()
+                .map(entry -> {
+                    Key keyOfFirstRecord = Key.create(entry.getValue().get(0).getValues(schema.getRowKeyFieldNames()));
+                    return new AbstractMap.SimpleEntry<>(partitionTree.getLeafPartition(keyOfFirstRecord).getId(), entry.getKey());
+                }).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        Map<Integer, List<FileInfo>> partitionNoToFileInfosMap = partitionIdToFileInfosMap.entrySet().stream()
+                .collect(Collectors.toMap(
+                        entry -> partitionIdToPartitionNoMap.get(entry.getKey()),
+                        Map.Entry::getValue));
+
+        int expectedTotalNoOfFiles = partitionNoToExpectedNoOfFilesMap.values().stream()
+                .mapToInt(Integer::valueOf)
+                .sum();
+
+        Set<Integer> allPartitionNoSet = Stream.of(
+                        partitionNoToFileInfosMap.keySet().stream(),
+                        partitionNoToExpectedNoOfFilesMap.keySet().stream(),
+                        partitionNoToExpectedRecordsMap.keySet().stream())
+                .flatMap(Function.identity())
+                .collect(Collectors.toSet());
+
+        assertThat(stateStore.getActiveFiles()).hasSize(expectedTotalNoOfFiles);
+        assertThat(allPartitionNoSet).allMatch(partitionNoToExpectedNoOfFilesMap::containsKey);
+
+        allPartitionNoSet.forEach(partitionNo -> verifyPartition(
+                partitionNoToFileInfosMap.getOrDefault(partitionNo, Collections.emptyList()),
+                partitionNoToExpectedNoOfFilesMap.get(partitionNo),
+                partitionNoToExpectedRecordsMap.getOrDefault(partitionNo, Collections.emptyList())));
+    }
+
+    private void verifyPartition(List<FileInfo> partitionFileInfoList,
+                                 int expectedNoOfFiles,
+                                 List<Record> expectedRecords) {
+        List<Record> actualRecords = readMergedRecordsFromPartitionDataFiles(schema, partitionFileInfoList, hadoopConfiguration);
+
+        assertThat(partitionFileInfoList).hasSize(expectedNoOfFiles); // Asserts that the partitionFileInfoList has the same size as the expected number of files
+        assertThat(actualRecords).containsExactlyInAnyOrderElementsOf(expectedRecords); // Asserts that the expected record list, when sorted, is identical to the record list in the partition
```

Comment:
We probably don't need these 2 comments, i think the code is descriptive enough and it clutters the line up a bit